optuna/samplers/_gp/sampler.py

```diff
@@ -184,18 +219,71 @@ def sample_relative(
             initial_kernel_params=self._kernel_params_cache,
             deterministic_objective=self._deterministic,
         )
+        constraints_kernel_params = []
+        for i, vals in enumerate(standardized_constraint_vals.T):
+            cache = (
+                self._constraints_kernel_params_cache and self._constraints_kernel_params_cache[i]
+            )
+            assert isinstance(cache, (gp.KernelParamsTensor)) or cache is None
+            constraints_kernel_params.append(
+                gp.fit_kernel_params(
+                    X=normalized_params,
+                    Y=vals,
+                    is_categorical=(
+                        internal_search_space.scale_types == gp_search_space.ScaleType.CATEGORICAL
+                    ),
+                    log_prior=self._log_prior,
+                    minimum_noise=self._minimum_noise,
+                    initial_kernel_params=cache,
+                    deterministic_objective=self._deterministic,
+                )
+            )
+
         self._kernel_params_cache = kernel_params
+        self._constraints_kernel_params_cache = constraints_kernel_params
 
+        is_feasible = np.all(constraint_vals <= 0, axis=1)
+        is_all_infeasible = not np.any(is_feasible)
+        # TODO(kAIto47802): If is_all_infeasible, the acquisition function for the objective
+        # function is ignored, so skipping the computation of kernel_params and acqf_params
+        # can improve speed.
         acqf_params = acqf.create_acqf_params(
             acqf_type=acqf.AcquisitionFunctionType.LOG_EI,
             kernel_params=kernel_params,
             search_space=internal_search_space,
             X=normalized_params,
-            Y=standarized_score_vals,
+            Y=standardized_score_vals,
+            max_Y=-np.inf if is_all_infeasible else np.max(standardized_score_vals[is_feasible]),
+        )
+        constraints_acqf_params = [
+            acqf.create_acqf_params(
+                acqf_type=acqf.AcquisitionFunctionType.LOG_PI,
+                kernel_params=params,
+                search_space=internal_search_space,
+                X=normalized_params,
+                Y=vals,
+                # Since 0 is the threshold value, we use the normalized value of 0.
+                max_Y=-mean / max(1e-10, std),
+            )
+            for params, vals, mean, std in zip(
+                constraints_kernel_params,
+                standardized_constraint_vals.T,
+                constraint_vals_mean,
+                constraint_vals_std,
+            )
+        ]
```

Comment:
Could you create a method that returns `constraints_acqf_params` and `is_feasible` instead of nest this process in `sample_relative`?